File: server/kotlin/src/main/kotlin/com/lebonpoint/application/usecases/GetItemUseCase.kt
package com.lebonpoint.application.usecases

import com.lebonpoint.application.services.ValidationService
import com.lebonpoint.domain.entities.Item
import com.lebonpoint.domain.repositories.ItemRepository
import com.lebonpoint.shared.NotFoundException

/**
 * Use case for getting an item by ID
 */
class GetItemUseCase(
    private val repository: ItemRepository
) {
    /**
     * Execute the use case
     * @param id Item ID
     * @return Item if found
     * @throws NotFoundException if item not found
     */
    suspend fun execute(id: Int): Item {
        ValidationService.validateId(id)

        return repository.findById(id)
            ?: throw NotFoundException("Item", id)
    }
}

File: server/typescript/src/application/use-cases/get-item.use-case.ts
import type { IItemRepository } from '../../domain/repositories/item.repository.port.js';
import { NotFoundError } from '../../shared/errors.js';
import type { Item } from '../../domain/entities/index.js';

/**
 * Get item use case
 */
export class GetItemUseCase {
  constructor(private readonly itemRepository: IItemRepository) {}

  /**
   * Execute the use case
   */
  async execute(id: number): Promise<Item> {
    const item = await this.itemRepository.findById(id);

    if (!item) {
      throw new NotFoundError('Item', id);
    }

    return item;
  }
}
